xdsl/backend/riscv/lowering/lower_func_riscv_func.py

```diff
@@ -0,0 +1,54 @@
+from xdsl.dialects import func, riscv_func
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+class LowerFuncOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: func.FuncOp, rewriter: PatternRewriter):
+        name = op.sym_name.data
+
+        if name != "main":
+            raise NotImplementedError("Only support lowering main function for now")
+
+        if op.body.blocks[0].args:
+            raise NotImplementedError(
+                "Only support functions with no arguments for now"
+            )
+
+        new_func = riscv_func.FuncOp(
+            op.sym_name.data, rewriter.move_region_contents_to_new_regions(op.body)
+        )
+
+        rewriter.replace_matched_op(new_func)
+
+
+class LowerFuncCallOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: func.Call, rewriter: PatternRewriter) -> None:
+        raise NotImplementedError("Function call lowering not implemented yet")
```

Comment:
Not sure why this is even in the PR, it just does absolutely nothing?

Reply:
I like it, it's a nice stub for us to replace later, and then the failing unit tests are replaced with working unit tests, so we get a little win in the future also